src/main/java/com/google/api/codegen/transformer/java/JavaGapicSurfaceTestTransformer.java

```diff
@@ -314,14 +354,15 @@ private GapicSurfaceTestCaseView createTestCaseView(
             .resourcesVarName(namer.localVarName(Name.from("resources")))
             .build());
 
-    if (methodContext.getFeatureConfig().useResourceNameFormatOption(resourcesField)) {
-      String resourceName = ResourceNameUtil.getResourceName(resourcesField);
-      Name resourceNameName = Name.upperCamel(resourceName);
+    if (methodContext.getFeatureConfig().useResourceNameFormatOption(resourcesFieldConfig)) {
+      String resourceClassName =
+          namer.getResourceTypeClassName(resourcesFieldConfig.getEntityName());
+      Name resourceName = namer.getResourceTypeName(resourcesFieldConfig.getEntityName());
       resourceTypeName =
           methodContext
               .getTypeTable()
               .getAndSaveNicknameForTypedResourceName(
-                  resourcesField, resourcesField.getType().makeOptional(), resourceName);
+                  resourcesField, resourcesField.getType().makeOptional(), resourceClassName);
```

Comment:
I think I would prefer to just pass the entity into this method so that we don't need the dual getResourceTypeName/getResourceTypeClassName functions. I would prefer to have only getResourceTypeName which returns the class name.